### PR TITLE
Closes #295 template documentation@devel

### LINKED ISSUE #295: Documentation: Create guidance around making templates
We do not provide any guidance to extension packages or to core team around creating a template for ADaMs. 

## Prep Work

- [ ] Review templates in core and extension packages for potential best practices
- [ ] Review templates in core and extension packages for not so great practices
- [ ] Sketch out Best Practices Guidance on Template creation and present to core team
- [ ] Discuss with core team best location for guidance (section in one vignette or separate vignette)
 
## Definition of Done
- [ ] Write a new vignette (or section of existing vignette) with guidance for creating a template.  

Quote from referenced issue #293 :
>In principle, we should also add something to the templates guidance but we do not have any. Should we add one?
> 
> What do you mean by template guidance?

We provide several templates in `inst/templates` but we do not have a guidance for these files, e.g., we do not describe when templates should be created, how they should be structured, which data they should use, ...

_Originally posted by @bundfussr in https://github.com/pharmaverse/admiraldev/issues/293#issuecomment-1619963647_

### PR DESCRIPTION
**Pull request by manciniedoardo**
Closes #295 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@bms63 @rossfarrugia once this PR is approved, I will quickly make a PR in admiraltemplate to link to this page in the README.

File: vignettes/package_extensions.Rmd
Comment (bms63) [inline vignettes/package_extensions.Rmd:57]: I think this might need to change to Dean Grundy.

File: vignettes/package_extensions.Rmd
Comment (bms63) [inline vignettes/package_extensions.Rmd:35]: I thought this stuff was just for the readme?

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:57]: Yes sounds good, do you have an email please?

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:35]: Sorry, you are probably right. As you know the text in this page was originally copied from the admiraltemplate README and I needed this `dyn_link` stuff for it to work - do you know how i should be linking to other pages instead? At the moment if I removed this entirely, calls such as the one on line 63 fail. Thanks.

**Comment by manciniedoardo** [comment]
[Corresponding PR in admiraldev](https://github.com/pharmaverse/admiraltemplate/pull/62).

File: vignettes/package_extensions.Rmd
Comment (bms63) [reply vignettes/package_extensions.Rmd:57]: On second thought - why not just asked them to reach out to the slack channel and remove the emails.  

Or ask them to reach out to the package maintainer - but not provide the name. 

Just trying to not have to update this section if roles change, people move on, etc.

File: vignettes/package_extensions.Rmd
Comment (bms63) [inline vignettes/package_extensions.Rmd:64]:
```suggestion
1. Each company should start to identify the required developer resources. Then each developer is required to complete the `{admiral`} [dummy issue for onboarding](https://github.com/pharmaverse/admiral/issues/1839), as well as reading up on the [admiraldev](https://pharmaverse.github.io/admiraldev/devel/) - especially the developer guides, which all need to be followed for package extensions.
1. Optionally it can be useful to host a kick-off meeting to decide how the team will work, for which we recommend agile/scrum practices.
```

File: vignettes/package_extensions.Rmd
Comment (bms63) [inline vignettes/package_extensions.Rmd:1]: this `dyn_link` is to help with making the links in the README.md and the front page of the packgedown site - since this is now a pkgdown article/user guide all of these links should have the dyn_link removed as we don't do it this way in the other user guides.

Comment (bms63) [inline vignettes/package_extensions.Rmd:1]: I like how detailed this is, but I feel like we need a highlights section or some type of graphic included.  perhaps a later issue, but it is a wall of text that is pretty hard to look at.

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [reply vignettes/package_extensions.Rmd:35]: fine to remove this section and update any links mentioned below to just use the main release website links, now that this is in admiraldev is the first in the dependency chain

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [reply vignettes/package_extensions.Rmd:57]: yes that works! any of us could then pick up such requests and follow up.
@bms63 please add Dean to our future sponsors/leads meetings if that gets confirmed. i spoke to him a while back and we worked together for years before at Roche so great to collaborate again if so

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [inline vignettes/package_extensions.Rmd:65]: suggest PHUSE MS teams now? seems most companies can access this

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [inline vignettes/package_extensions.Rmd:91]: add `templates`, and remove `test data` - as think that'd confuse with pharmaversesdtm/adam

**Review by rossfarrugia — CHANGES_REQUESTED** [review]

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:57]: Agree, changed.

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:35]: updated links, thank you

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:64]: implemented, thank you

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:65]: updated, thanks.

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:91]: updated, thanks.

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [inline vignettes/package_extensions.Rmd:31]: admiraldev links usually use `devel` version as default as we always want people to refer to the very latest guidance so update link to https://pharmaverse.github.io/admiraldev/devel/index.html

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [inline vignettes/package_extensions.Rmd:33]:
```suggestion
1. Set up a "admiralxxx_dev" channel on Slack to add all team members to for informal team chat, and agree a way to share working documents across the co-development team. We recommend to use a new folder on the pharmaverse MS teams - Michael Rimler (michael.s.rimler@gsk.com) could help with this as our rep on the PHUSE board.
```

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [inline vignettes/package_extensions.Rmd:34]: i would delete this in favour of what i added above - the link wouldn't work for people without access anyway, so Michael would get them on to this

File: vignettes/package_extensions.Rmd
Comment (rossfarrugia) [reply vignettes/package_extensions.Rmd:31]: same comment throughout for any other occurrences

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd:34]: updated

File: vignettes/package_extensions.Rmd
Comment (manciniedoardo) [reply vignettes/package_extensions.Rmd]: updated

**Review by rossfarrugia — APPROVED** [review]

**Review by zdz2101 — APPROVED** [review]
Looks good to me too!

**Review by bms63 — APPROVED** [review]

**Pull request merged**
